src/animationBuilder.tsx

```diff
@@ -42,21 +83,13 @@ export function maybeBuild(
   if (isAnimationBuilder(layoutAnimationOrBuilder)) {
     const animationFactory = layoutAnimationOrBuilder.build();
     const layoutAnimation = animationFactory(mockTargetValues);
-    const animatedStyle = layoutAnimation.animations;
-
-    const getCommonProperties = (obj1: object, obj2: object) =>
-      Object.keys(obj1).filter((key) =>
-        Object.prototype.hasOwnProperty.call(obj2, key)
-      );
+    const layoutAnimationStyle = layoutAnimation.animations;
 
-    const commonProperties = getCommonProperties(animatedStyle, style || {});
-    if (commonProperties.length > 0) {
-      console.warn(
-        `[Reanimated] ${
-          commonProperties.length === 1 ? 'Property' : 'Properties: '
-        } "${commonProperties.join(
-          ', '
-        )}" of ${displayName} may be overwritten with layout animation. Please create a wrapper with the layout animation you want to apply.`
+    if (__DEV__) {
+      maybeReportOverwrittenProperties(
+        layoutAnimationStyle,
+        style || {},
+        displayName
```

Comment:
Does it make sense to call `maybeReportOverwrittenProperties` for an empty object if `style` is undefined?
```suggestion
    if (__DEV__ && style) {
      maybeReportOverwrittenProperties(
        layoutAnimationStyle,
        style,
        displayName
```